x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/search-business-rules/10_pinned_query.yml

```diff
@@ -78,3 +78,30 @@ setup:
   - match: { hits.hits.2._index: "test1" }
   - match: { hits.hits.3._id: "b" }
 
+---
+"Test pinned query with docs":
```

Comment:
I'm not sure if this is possible with yaml tests, but should we have a previous version yaml test that verifies it's a bad request if we don't include the index?

Reply:
Is there any good documentation about writing yaml tests?

Reply:
https://github.com/elastic/elasticsearch/blob/main/TESTING.asciidoc#testing-the-rest-layer

Reply:
> but should we have a previous version yaml test that verifies it's a bad request if we don't include the index?

You can do this by adjusting `skip`. 

So, something like 

```
  - skip:
      version: "8.10.99 - " // Skip the latest version as its now optional
      reason: "'_index' was made optional in 8.11, this test is for explicit failure tests"
  //my test that asserts it was explicit
```

We honestly should have had a test that verified it was required before :/ I am not sure why we didn't.

Reply:
I added a test for that 👍